src/OpenFeature.Providers.Ofrep/OfrepProvider.cs

```diff
@@ -0,0 +1,186 @@
+        CancellationToken cancellationToken = default) =>
+        this.ResolveFlag(flagKey, defaultValue, context,
+            cancellationToken);
+
+    /// <inheritdoc/>
+    public override Task<ResolutionDetails<int>> ResolveIntegerValueAsync(
+        string flagKey, int defaultValue, EvaluationContext? context = null,
+        CancellationToken cancellationToken = default) =>
+        this.ResolveFlag(flagKey, defaultValue, context,
+            cancellationToken);
+
+    /// <inheritdoc/>
+    public override Task<ResolutionDetails<double>> ResolveDoubleValueAsync(
+        string flagKey, double defaultValue, EvaluationContext? context = null,
+        CancellationToken cancellationToken = default) =>
+        this.ResolveFlag(flagKey, defaultValue, context,
+            cancellationToken);
+
+    /// <inheritdoc/>
+    public override async Task<ResolutionDetails<Value>>
+        ResolveStructureValueAsync(string flagKey, Value defaultValue,
+            EvaluationContext? context = null,
+            CancellationToken cancellationToken = default)
+
+    {
+#if NET8_0_OR_GREATER
+        ArgumentException.ThrowIfNullOrWhiteSpace(flagKey);
+#else
+        if (flagKey == null)
+        {
+            throw new ArgumentNullException(nameof(flagKey));
+        }
+#endif
+
+        var response =
+            await this._client.EvaluateFlag(flagKey, defaultValue.AsObject,
+                context, cancellationToken).ConfigureAwait(false);
+
+        return new ResolutionDetails<Value>(
+            flagKey,
+            response.Value != null ? new Value(response.Value) : new Value(String.Empty),
+            MapErrorType(response.ErrorCode ?? string.Empty),
+            reason: response.Reason,
+            variant: response.Variant,
+            errorMessage: response.ErrorMessage,
+            flagMetadata: response.Metadata != null ? new ImmutableMetadata(response.Metadata) : null);
+    }
+
+    /// <summary>
+    /// Resolves a flag of the specified type using the OFREP client.
+    /// </summary>
+    /// <typeparam name="T">The type of the flag value</typeparam>
+    /// <param name="flagKey">The unique identifier for the flag</param>
+    /// <param name="defaultValue">The default value to return if the flag cannot be resolved</param>
+    /// <param name="context">Optional evaluation context with targeting information</param>
+    /// <param name="cancellationToken">A token to cancel the operation</param>
+    /// <returns>Resolution details containing the flag value and
+    /// metadata</returns>
+    private async Task<ResolutionDetails<T>> ResolveFlag<T>(
+        string flagKey,
+        T defaultValue,
+        EvaluationContext? context,
+        CancellationToken cancellationToken)
+    {
+#if NET8_0_OR_GREATER
+        ArgumentNullException.ThrowIfNull(flagKey);
+#else
+        if (flagKey == null)
+        {
+            throw new ArgumentNullException(nameof(flagKey));
+        }
+#endif
```

Comment:
If this is just to make use of a language feature, I would say vote to remove the compiler directive and just use the common version. I think adding compiler directives like this takes away from readability so I recommend only doing it when we really have to.

If we "really have to" here for some reason I don't understand, that's OK.